backend/Tournament/managers.py

```diff
@@ -0,0 +1,174 @@
+from django.db import transaction
+from django.utils import timezone
+from datetime import timedelta
+
+from .models import Tournament, \
+                        TournamentPlayer
+from .serializers import TournamentCreationSerializer
+from .exceptions import TournamentSetupException, \
+                            TournamentInProgressException
+from .tournamentState import TournamentState
+from .tournamentMacros import TOURNAMENT_EXPIRY_TIME_SECONDS
+from User.models import User
+from Match.models import Match
+from Match.matchState import MatchState
+from Player.models import Player
+
+import sys
+
+class TournamentSetupManager:
+    @staticmethod
+    def create_tournament_and_tournament_player_for_host(validated_data: TournamentCreationSerializer.validated_data):
+        try:
+            with transaction.atomic():
+                tournament = Tournament.objects.create(
+                    name=validated_data['name'] if validated_data['name'] else None,
+                    host_user=validated_data['host_user'],
+                    player_amount=validated_data['player_amount'],
+                    expires_ts=timezone.now() + timedelta(TOURNAMENT_EXPIRY_TIME_SECONDS),
+                )
+
+                TournamentPlayer.objects.create(
+                    tournament=tournament,
+                    user=tournament.host_user,
+                    display_name=validated_data['host_user_display_name'] if validated_data['host_user_display_name'] else None,
+                )
+
+                return tournament
+
+        except Exception as e:
+            raise TournamentSetupException(e)
+
+    @staticmethod
+    def setup_tournament_matches(tournament: Tournament) -> None:        
+        if tournament.state != TournamentState.LOBBY or tournament.next_match != 0:
+            raise TournamentSetupException("Tournament is not in a valid state to set up matchups.")
+        
+        tournament_players = list(tournament.players.all())
+        if len(tournament_players) != tournament.player_amount:
+            raise TournamentSetupException("The number of tournament_players does not match the expected player amount.")
+        
+        match_amount = tournament.player_amount - 1
```

Comment:
Is this dangerous, can the player amount be less than 1?

Reply:
no because in the validation of the tournament creation serializer we check that the player_amount is either 4 or 8.

Reply:
Alright, then we just hope we dont forget this and try to save some possible invalid value in the db!